packages/camera/camera_android_camerax/test/android_camera_camerax_test.dart

```diff
@@ -405,17 +418,132 @@ void main() {
+    when(mockPlane.getRowStride())
+        .thenAnswer((_) => Future<int>.value(rowStride));
+    when(mockPlane.getPixelStride())
+        .thenAnswer((_) => Future<int>.value(pixelStride));
+    when(mockImageProxy.getFormat())
+        .thenAnswer((_) => Future<int>.value(imageFormat));
+    when(mockImageProxy.getHeight())
+        .thenAnswer((_) => Future<int>.value(imageHeight));
+    when(mockImageProxy.getWidth())
+        .thenAnswer((_) => Future<int>.value(imageWidth));
+
+    final StreamSubscription<CameraImageData>
+        onStreamedFrameAvailableSubscription =
+        camera.onStreamedFrameAvailable(22).listen((CameraImageData imageData) {
+      // Test Analyzer correctly process ImageProxy instances.
+      expect(imageData.planes.length, equals(0));
+      expect(imageData.planes[0].bytes, equals(buffer));
+      expect(imageData.planes[0].bytesPerRow, equals(rowStride));
+      expect(imageData.planes[0].bytesPerPixel, equals(pixelStride));
+      expect(imageData.format.raw, equals(imageFormat));
+      expect(imageData.height, equals(imageHeight));
+      expect(imageData.width, equals(imageWidth));
+    });
+
+    // Test ImageAnalysis use case is bound to ProcessCameraProvider.
+    final Analyzer capturedAnalyzer =
+        verify(camera.mockImageAnalysis.setAnalyzer(captureAny)).captured.single
+            as Analyzer;
+    verify(mockProcessCameraProvider.bindToLifecycle(
+        mockCameraSelector, <UseCase>[camera.mockImageAnalysis]));
+
+    await capturedAnalyzer.analyze(mockImageProxy);
+    onStreamedFrameAvailableSubscription.cancel();
+  });
+
+  test(
+      'onStreamedFrameAvaiable returns stream that responds expectedly to being canceled',
+      () async {
+    final MockAndroidCameraCameraX camera = MockAndroidCameraCameraX();
+    final ProcessCameraProvider mockProcessCameraProvider =
+        MockProcessCameraProvider();
+    final CameraSelector mockCameraSelector = MockCameraSelector();
+    final Camera mockCamera = MockCamera();
+
+    camera.processCameraProvider = mockProcessCameraProvider;
+    camera.cameraSelector = mockCameraSelector;
+    camera.createDetachedCallbacks = true;
+
+    when(mockProcessCameraProvider.bindToLifecycle(
+            mockCameraSelector, <UseCase>[camera.mockImageAnalysis]))
+        .thenAnswer((_) async => mockCamera);
+
+    final StreamSubscription<CameraImageData> imageStreamSubscription =
+        camera.onStreamedFrameAvailable(32).listen((CameraImageData data) {});
+
+    when(mockProcessCameraProvider.isBound(camera.mockImageAnalysis))
+        .thenAnswer((_) async => Future<bool>.value(true));
+
+    await imageStreamSubscription.cancel();
+
+    verify(camera.mockImageAnalysis.clearAnalyzer());
+  });
 }
 
 /// Mock of [AndroidCameraCameraX] that stubs behavior of some methods for
 /// testing.
-class MockAndroidCameraCamerax extends AndroidCameraCameraX {
+class MockAndroidCameraCameraX extends AndroidCameraCameraX {
```

Comment:
Would any other classes benefit from this mock? if so moving it to its own class might be helpful.